cmd/mayactl/app/command/volume_info.go

```diff
@@ -68,33 +92,38 @@ func NewCmdVolumeInfo() *cobra.Command {
 
 // RunVolumeInfo runs info command and make call to DisplayVolumeInfo
 func (c *CmdVolumeOptions) RunVolumeInfo(cmd *cobra.Command) error {
-	annotation := &Annotations{}
-	// GetVolumeAnnotation is called to get the volume controller's info such as
+	volumeInfo := &v1alpha1.CASVolume{}
+	// FetchVolumeInfo is called to get the volume controller's info such as
 	// controller's IP, status, iqn, replica IPs etc.
-	err := annotation.GetVolAnnotations(c.volName, c.namespace)
+	err := volumeInfo.FetchVolumeInfo(mapiserver.GetURL()+listPath+c.volName, c.volName, c.namespace)
 	if err != nil {
-		return nil
+		return err
 	}
 
-	ctrlstatus := strings.Split(annotation.ControllerStatus, ",")
-	for i := range ctrlstatus {
-		if ctrlstatus[i] != "running" {
-			fmt.Printf("Unable to fetch volume details, Volume controller's status is '%s'.\n", annotation.ControllerStatus)
-			return nil
+	// Initiallize an instance of ReplicaCollection, json response recieved from the
+	collection := client.ReplicaCollection{}
+	if volumeInfo.GetField("CasType") == "jiva" {
```

Comment:
Can be ensure that `"jiva"` is not hardcoded.

Reply:
sure